### PR TITLE
Transmission: Update to 4.0.6

### LINKED ISSUE #6146: Transmission update
### Software Name

Transmission

### Brief Description

Please update transmission software.
Thank you.

### Website

https://transmissionbt.com/

### Documentation

_No response_

### Build/Installation documentation

https://github.com/transmission/transmission/blob/main/docs/Building-Transmission.md

### Source code

https://github.com/transmission/transmission

### OpenSource License

GPLv3

### Other License

_No response_

### PR DESCRIPTION
**Pull request by mreid-tt**
## Description

This pull request includes the following:
1. Update to Transmission v4.0.6 (bugfix-only release)
2. Include patched source for missing linked submodules

Fixes #6146

## Checklist

- [x] Build rule `all-supported` completed successfully
- [x] New installation of package completed successfully
- [x] Package upgrade completed successfully (Manually install the package again)
- [x] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Package-Update-Policy#tests-checks)
- [x] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created

### Type of change

- [x] Package update

### CODE REVIEW

File: cross/c-ares/Makefile [hunk 1]
```diff
@@ -2,7 +2,8 @@ PKG_NAME = c-ares
 PKG_VERS = 1.27.0
 PKG_EXT = tar.gz
 PKG_DIST_NAME = $(PKG_NAME)-$(PKG_VERS).$(PKG_EXT)
-PKG_DIST_SITE = https://c-ares.org/download
+PKG_DIST_DIR = $(subst ., _ ,$(subst -,,$(PKG_NAME))-$(PKG_VERS))
+PKG_DIST_SITE = https://github.com/c-ares/$(PKG_NAME)/releases/download/${PKG_DIST_DIR}
```
Comment (hgy59) [inline cross/c-ares/Makefile:6]: If you need to touch this makefile, it is probably worth an update to the latest version (1.29)

and please keep the code simple (and readable) like
`PKG_DIST_DIR = cares-$(subst ., _ ,$(PKG_VERS))`

_OR_ in one line:
`PKG_DIST_SITE = https://github.com/c-ares/c-ares/releases/download/cares-$(subst ., _ ,$(PKG_VERS))`

**Comment by hgy59** [comment]
@mreid-tt If you don't mind, I will create a dedicated PR for c-ares update
it is an issue in #6127 too (build of rutorren) and needs older version for OLD_PPC_ARCHS

**Comment by mreid-tt** [comment]
> @mreid-tt If you don't mind, I will create a dedicated PR for c-ares update it is an issue in #6127 too (build of rutorrent) and needs an older version for OLD_PPC_ARCHS

Sure @hgy59 not a problem. I'll remove the c-ares code and wait until you merge so I can re-base.

**Comment by mreid-tt** [comment]
Seems we are unable to build due to this bug: https://github.com/transmission/transmission/issues/6873.

**Comment by hgy59** [comment]
> Seems we are unable to build due to this bug: [transmission/transmission#6873](https://github.com/transmission/transmission/issues/6873).

@mreid-tt I have created a source archive that includes the submodules and added it to synocommunity releases in https://github.com/SynoCommunity/spksrc/releases/tag/transmission

I first tried to enhance the spksrc framework to use git repositories with submodules, but since the `PKG_DOWNLOAD_METHOD=git` creates different hashes (digests file) when recreating the archive with unmodified content, this is not a solution.

I also provided the script to create the archive for documentation and content validation.

**Comment by mreid-tt** [comment]
@hgy59 appreciate your workaround to this issue. I'll begin package testing and then move to publishing.

**Pull request merged**